src/database/sqlite.rs

```diff
@@ -467,7 +467,7 @@ impl SqliteDatabase {
         &self,
         txid: &[u8],
     ) -> Result<Option<TransactionDetails>, Error> {
-        let mut statement = self.connection.prepare_cached("SELECT transaction_details.timestamp, transaction_details.received, transaction_details.sent, transaction_details.fee, transaction_details.height, transactions.raw_tx FROM transaction_details, transactions WHERE transaction_details.txid=transactions.txid AND transaction_details.txid=:txid")?;
```

Comment:
I would keep this query the same as it was before, because while it doesn't make too much difference for sqlite I think it would be a lot slower to make two queries over network rather than one. So if in the future we add support for postgres or mysql it will probably be based on the sqlite code, and for this reason I would try to be forward-thinking here.

Reply:
If it makes the code cleaner, we might decide to pass a `include_raw` parameter to `select_transaction_details_by_txid` and, even though we will still fetch the transaction from the db using this query, decide not to return it if it's not needed

Reply:
I will go with just setting the transaction field of the return object to None.